### ISSUE TITLE
[R2R] mainv1 mainv2 StructABI Test Failures

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#23643]
Looks like #22560 regressed these R2R tests on Windows_NT x86 and Windows_NT x64 (https://dev.azure.com/dnceng/public/_build/results?buildId=139566)

**readytorun_tests._mainv1_mainv1_._mainv1_mainv1_cmd**
```
Error compiling mainv1.exe: Unspecified error (0x80004005 (E_FAIL))
Error: compilation failed for \"mainv1.exe\" (0x80004005)
Error compiling mainv1.org: Unspecified error (0x80004005 (E_FAIL))
Error: compilation failed for \"mainv1.org\" (0x80004005)


Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\Reports\\readytorun.tests\\mainv1\\mainv1.output.txt
Raw output:
BEGIN EXECUTION
        1 file(s) copied.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image test.ni.dll generated successfully.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image fieldgetter.ni.dll generated successfully.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

        1 file(s) copied.
\" C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload\\crossgen.exe\"  /Platform_Assemblies_Paths C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv1\\IL;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv1 /in mainv1.org /out mainv1.exe
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Crossgen failed with exitcode - -2147467259
Test Harness Exitcode is : 1

To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload
> C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv1\\mainv1.cmd

Expected: True
Actual:   False
```
**readytorun_tests._mainv2_mainv2_._mainv2_mainv2_cmd**
```
Error compiling mainv2.exe: Unspecified error (0x80004005 (E_FAIL))
Error: compilation failed for \"mainv2.exe\" (0x80004005)
Error compiling mainv2.org: Unspecified error (0x80004005 (E_FAIL))
Error: compilation failed for \"mainv2.org\" (0x80004005)


Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\Reports\\readytorun.tests\\mainv2\\mainv2.output.txt
Raw output:
BEGIN EXECUTION
        1 file(s) copied.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image fieldgetter.ni.dll generated successfully.
        1 file(s) copied.
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image test.ni.dll generated successfully.
        1 file(s) copied.
\" C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload\\crossgen.exe\"  /Platform_Assemblies_Paths C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv2\\IL;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv2 /in mainv2.org /out mainv2.exe
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Crossgen failed with exitcode - -2147467259
Test Harness Exitcode is : 1

To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload
> C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\c250980e-daeb-4533-aee3-8bd74e3ea370\\Exec\\mainv2\\mainv2.cmd

Expected: True
Actual:   False
```
**JIT_Directed._StructABI_StructABI_StructABI_._StructABI_StructABI_StructABI_cmd()**
```
Assert failure(PID 4140 [0x0000102c], Thread: 2492 [0x09bc]): !IsTypeDesc()

CROSSGEN! <no symbol> + 0x0 (0x00007ff6`0fcfe44e)
CROSSGEN! <no symbol> + 0x0 (0x00007ff6`0fd0984f)
CROSSGEN! <no symbol> + 0x0 (0x00007ff6`0fd863d7)
CROSSGEN! <no symbol> + 0x0 (0x00007ff6`0fd6d05f)
CROSSGEN! <no symbol> + 0x0 (0x00007ff6`0fd798e2)
CLRJIT! sxsJitStartup + 0x593B8 (0x00007fff`3fccc468)
CLRJIT! sxsJitStartup + 0x696FD (0x00007fff`3fcdc7ad)
CLRJIT! sxsJitStartup + 0x62763 (0x00007fff`3fcd5813)
CLRJIT! sxsJitStartup + 0x5C0CD (0x00007fff`3fccf17d)
CLRJIT! sxsJitStartup + 0x5B7E1 (0x00007fff`3fcce891)
    File: f:\\workspace\\_work\\1\\s\\src\\vm\\typehandle.inl Line: 115
    Image: C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload\\crossgen.exe



Return code:      1
Raw output file:      C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\7d22399e-12c2-4eb5-a699-120e64a7247b\\Exec\\Reports\\JIT.Directed\\StructABI\\StructABI\\StructABI.output.txt
Raw output:
BEGIN EXECUTION
        1 file(s) copied.
\" C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload\\crossgen.exe\"  /Platform_Assemblies_Paths C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\7d22399e-12c2-4eb5-a699-120e64a7247b\\Exec\\StructABI\\StructABI\\IL;C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\7d22399e-12c2-4eb5-a699-120e64a7247b\\Exec\\StructABI\\StructABI /in StructABI.org /out StructABI.exe
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Crossgen failed with exitcode - -1073740286
Test Harness Exitcode is : 1

To run the test:
> set CORE_ROOT=C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Payload
> C:\\dotnetbuild\\work\\61fed898-2eb9-453e-a1ae-60d5c255b5f1\\Work\\7d22399e-12c2-4eb5-a699-120e64a7247b\\Exec\\StructABI\\StructABI\\StructABI.cmd

Expected: True
Actual:   False
```

cc @fadimounir @jkotas

**Comment by fadimounir** [comment]
Taking a look

**Comment by fadimounir** [comment]
Fix: #23828
